Code/Tools/AssetProcessor/native/tests/utilities/StatsCaptureTest.cpp

```diff
@@ -0,0 +1,200 @@
+// Double-intiailize is an error
+TEST_F(AssetProcessorTest, StatsCaptureTest_DoubleInitializeIsAnAssert)
+{
+    m_errorAbsorber->Clear();
+    
+    AssetProcessor::StatsCapture::Initialize();
+    AssetProcessor::StatsCapture::Initialize();
+    
+    EXPECT_EQ(m_errorAbsorber->m_numErrorsAbsorbed, 0);
+    EXPECT_EQ(m_errorAbsorber->m_numAssertsAbsorbed, 1); // not allowed to assert on this
+
+    AssetProcessor::StatsCapture::BeginCaptureStat("Test");
+    AssetProcessor::StatsCapture::Shutdown();
+}
+
+class StatsCaptureOutputTest : public AssetProcessorTest, public AZ::Debug::TraceMessageBus::Handler
+{
+public:
+    void SetUp() override
+    {
+        AssetProcessorTest::SetUp();
+        AssetProcessor::StatsCapture::Initialize();
+    }
+    
+    // dump but also capture the dump as a vector of lines:
+    void Dump()
+    {
+        AZ::Debug::TraceMessageBus::Handler::BusConnect();
+        AssetProcessor::StatsCapture::Dump();
+        AZ::Debug::TraceMessageBus::Handler::BusDisconnect();
+    }
+    
+    virtual bool OnPrintf(const char* /*window*/, const char* message)
+    {
+        m_gatheredMessages.emplace_back(message);
+        AZ::StringFunc::TrimWhiteSpace(m_gatheredMessages.back(), true, true);
+        return false;
+    }
+    
+    void TearDown() override
+    {
+        m_gatheredMessages = {};
+        
+        AssetProcessor::StatsCapture::Shutdown();
+        AssetProcessorTest::TearDown();
+    }
+    
+    AZStd::vector<AZStd::string> m_gatheredMessages;
+};
+
+// turning off machine and human readable mode, should not dump anything.
+TEST_F(StatsCaptureOutputTest, StatsCaptureTest_DisabledByRegset_DumpsNothing)
+{
+    auto registry = AZ::SettingsRegistry::Get();
+    ASSERT_NE(registry, nullptr);
+    registry->Set("/Amazon/AssetProcessor/Settings/Stats/HumanReadable", false);
+    registry->Set("/Amazon/AssetProcessor/Settings/Stats/MachineReadable", false);
+    AssetProcessor::StatsCapture::BeginCaptureStat("Test");
+    AssetProcessor::StatsCapture::EndCaptureStat("Test");
+    Dump();
+    EXPECT_EQ(m_gatheredMessages.size(), 0);
+}
+
+// turning on Human Readable, turn off Machine Readable, should not output any machine readable stats.
+TEST_F(StatsCaptureOutputTest, StatsCaptureTest_HumanReadableOnly_DumpsNoMachineReadable)
+{
+    auto registry = AZ::SettingsRegistry::Get();
+    ASSERT_NE(registry, nullptr);
+    registry->Set("/Amazon/AssetProcessor/Settings/Stats/HumanReadable", true);
+    registry->Set("/Amazon/AssetProcessor/Settings/Stats/MachineReadable", false);
+    AssetProcessor::StatsCapture::BeginCaptureStat("Test");
+    AssetProcessor::StatsCapture::EndCaptureStat("Test");
+    Dump();
+    EXPECT_GT(m_gatheredMessages.size(), 0);
+    for (const auto& message : m_gatheredMessages)
+    {
+        // we expect to see ZERO "Machine Readable" lines
+        EXPECT_EQ(message.find("MachineReadableStat:"), AZStd::string::npos) << "Found unexpected line in output: " << message.c_str();
```

Comment:
Future Reference: FYI our string implementations now has support for the [contains](https://en.cppreference.com/w/cpp/string/basic_string/contains) method. That would perform the comparison against npos and return a boolean

Reply:
> This is great timing, we were just discussing last week how we wanted Jenkins jobs that track asset processor performance so we can keep a better eye on it, and better quantify the benefits of performance improvements.

By making it machine readable I'm hoping we can just parse it as part of the cotinuous build system and keep metrics.

It runs on regset because regset allows you to override reg on command line, ie
```
AssetProcessorBatch.exe --project-path <path> --regset="/Amazon/AssetProcessor/Settings/Stats/MachineReadable=true" --regset="/Amazon/AssetProcessor/Settings/Stats/HumanReadable=false"   --regset="/Amazon/AssetProcessor/Settings/Stats/MaxCumulativeStats=50" 
```